src/ConfigInterface.php

```diff
@@ -4,6 +4,56 @@
 
 namespace Laminas\ServiceManager;
 
+use ArrayAccess;
+use Interop\Container\ContainerInterface;
+
+/**
+ * @see ContainerInterface
+ * @see ArrayAccess
+ *
+ * @psalm-type AbstractFactoriesConfigurationType = array<
+ *      array-key,
+ *      (class-string<Factory\AbstractFactoryInterface>|Factory\AbstractFactoryInterface)
+ * >
+ * @psalm-type DelegatorsConfigurationType = array<
+ *      string,
+ *      array<
+ *          array-key,
+ *          (class-string<Factory\DelegatorFactoryInterface>|Factory\DelegatorFactoryInterface)
+ *          |class-string
+ *          |callable(ContainerInterface,string,callable():object,array<mixed>):object
+ *          |(class-string<DelegatorFactoryInterface>|DelegatorFactoryInterface)
+ *      >
+ * >
+ * @psalm-type FactoriesConfigurationType = array<
+ *      string,
+ *      (class-string<Factory\FactoryInterface>|Factory\FactoryInterface)
+ *      |class-string
+ *      |callable(ContainerInterface,string,array<mixed>)
+ * >
+ * @psalm-type InitializersConfigurationType = array<
+ *      array-key,
+ *      (class-string<Initializer\InitializerInterface>|Initializer\InitializerInterface)
+ *      |callable(ContainerInterface,object):void
+ * >
+ * @psalm-type LazyServicesConfigurationType = array{
+ *      class_map?:array<string,string>,
+ *      proxies_namespace?:non-empty-string,
+ *      proxies_target_dir?:non-empty-string,
+ *      write_proxy_files?:bool
+ * }
+ * @psalm-type ServiceManagerConfigurationType = array{
+ *     abstract_factories?: AbstractFactoriesConfigurationType,
+ *     aliases?: array<string,string>,
+ *     delegators?: DelegatorsConfigurationType,
+ *     factories?: FactoriesConfigurationType,
+ *     initializers?: InitializersConfigurationType,
+ *     invokables?: array<string,string>,
+ *     lazy_services?: LazyServicesConfigurationType,
+ *     services?:array<string,object>&array{config?:ArrayAccess<string,mixed>|array<string,mixed>},
```

Comment:
https://github.com/laminas/laminas-servicemanager/pull/98/files#diff-c88e8922673ece50d04aef21f4dc95b87bd465868eb073480467f87e626e3b0aR478

This allows `array|object`.
Yes, I know that this would not only allow `config` but also other arrays.
As @Ocramius pointed out in #99, this might be a BC break (even tho, we still accept arrays), so should I change this to `array<string,object|array>` instead?